tests/framework/cli/micropkg/test_micropkg_pull.py

```diff
@@ -0,0 +1,732 @@
+
+        pipeline_name = alias or pipeline_name
+        source_dest = fake_package_path / "pipelines" / pipeline_name
+        test_dest = fake_repo_path / "src" / "tests" / "pipelines" / pipeline_name
+        config_env = env or "base"
+        dest_params_config = (
+            fake_repo_path
+            / settings.CONF_ROOT
+            / config_env
+            / "parameters"
+            / f"{pipeline_name}.yml"
+        )
+
+        assert not filecmp.dircmp(source_path, source_dest).diff_files
+        assert not filecmp.dircmp(test_path, test_dest).diff_files
+        assert source_params_config.read_bytes() == dest_params_config.read_bytes()
+
+    def test_pipeline_alias_refactors_imports(
+        self, fake_project_cli, fake_package_path, fake_repo_path, fake_metadata
+    ):
+        call_pipeline_create(fake_project_cli, fake_metadata)
+        pipeline_file = fake_package_path / "pipelines" / PIPELINE_NAME / "pipeline.py"
+        import_stmt = (
+            f"import {fake_metadata.package_name}.pipelines.{PIPELINE_NAME}.nodes"
+        )
+        with pipeline_file.open("a") as f:
+            f.write(import_stmt)
+
+        package_alias = "alpha"
+        pull_alias = "beta"
+
+        call_micropkg_package(
+            cli=fake_project_cli, metadata=fake_metadata, alias=package_alias
+        )
+
+        wheel_file = (
+            fake_repo_path
+            / "src"
+            / "dist"
+            / _get_wheel_name(name=package_alias, version="0.1")
+        )
+        CliRunner().invoke(
+            fake_project_cli, ["micropkg", "pull", str(wheel_file)], obj=fake_metadata
+        )
+        CliRunner().invoke(
+            fake_project_cli,
+            ["micropkg", "pull", str(wheel_file), "--alias", pull_alias],
+            obj=fake_metadata,
+        )
+
+        for alias in (package_alias, pull_alias):
+            path = fake_package_path / "pipelines" / alias / "pipeline.py"
+            file_content = path.read_text()
+            expected_stmt = (
+                f"import {fake_metadata.package_name}.pipelines.{alias}.nodes"
+            )
+            assert expected_stmt in file_content
+
+    def test_micropkg_pull_from_aliased_pipeline_conflicting_name(
+        self, fake_project_cli, fake_package_path, fake_repo_path, fake_metadata
+    ):
+        package_name = fake_metadata.package_name
+        call_pipeline_create(fake_project_cli, fake_metadata)
+        pipeline_file = fake_package_path / "pipelines" / PIPELINE_NAME / "pipeline.py"
+        import_stmt = f"import {package_name}.pipelines.{PIPELINE_NAME}.nodes"
+        with pipeline_file.open("a") as f:
+            f.write(import_stmt)
+
+        call_micropkg_package(
+            cli=fake_project_cli, metadata=fake_metadata, alias=package_name
+        )
+        wheel_file = (
+            fake_repo_path
+            / "src"
+            / "dist"
+            / _get_wheel_name(name=package_name, version="0.1")
```

Comment:
as mentioned before, these will turn into `sdist_file = ...` or something of the sort.